kolibri/core/errorreports/schemas.py

```diff
@@ -0,0 +1,54 @@
+context_frontend_schema = {
+    "type": "object",
+    "properties": {
+        "browser": {"type": "string"},
+        "component": {"type": "string", "optional": True},
+        "device": {
+            "type": "object",
+            "properties": {
+                "type": {"type": "string"},  # "desktop", "tablet", "mobile"
+                "platform": {
+                    "type": "string",
+                    "optional": True,
+                },  # "windows", "mac", "linux", "android", "ios"
+                "screen": {
+                    "type": "object",
+                    "properties": {
+                        "width": {"type": "integer"},
+                        "height": {"type": "integer"},
+                    },
+                },
+            },
+        },
+    },
+}
+default_context_frontend_schema = {"browser": "", "component": "N/A", "device": {}}
+context_backend_schema = {
+    "type": "object",
+    "properties": {
+        "request_info": {
+            "type": "object",
+            "properties": {
+                "url": {"type": "string", "optional": True},
+                "method": {"type": "string", "optional": True},
+                "headers": {"type": "object", "optional": True},
+                "body": {"type": "string", "optional": True},
```

Comment:
I think properties are optional by default so there is not need to explicitly set them as optional. So you might as well do away with the optional flag?

Reply:
Also, we can use the required property that specifies the required fields instead where necessary

Reply:
(There are other sections in the schema with the same)

Reply:
@akolson I too had a doubt about this. But this comment suggested otherwise: https://github.com/learningequality/kolibri/blob/dc7a8433f3633c75892171b2c4a2645f5998dbd4/kolibri/core/auth/models.py#L90

Reply:
Since the use of the `optional` property is specific to a particular version of jsonschema, we'll carry on with using it, but create a follow up issue to to update all jsonschemas as python 2.7 support has been dropped (the main reason the current version of jsonschema is used)